presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -101,16 +108,33 @@ public void reset()
         }
     }
 
-    public Type getType(int channel)
+    public static PageBuilder resetNew(PageBuilder other)
```

Comment:
Can we rename all of the `resetNew` methods to `newPageBuilderLike` and `newBlockBuilderLike`